var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -234,3 +234,31 @@
     line: '{{ item.key }}="{{ item.value }}"'
   with_items:
     - { key: 'JAVA_HOME', value: '/usr/lib/jvm/oracle-java8-jdk-amd64' }
+
+- shell: dpkg-query -Wf '${Package}' 'delphix-platform-*' | sed 's/delphix-platform-//'
```

Comment:
Should we create a script/command that encapsulated this code? Since it's used in the upgrade scripts as well, I'm just wondering if wrapping it would make things a little more maintainable.

Reply:
Yeah, I think that's reasonable. Alternatively, we could install a file that has the platform as its contents, if that would be simpler for consumers. I'm not sure where such a file would go on the filesystem though. Something along the lines of
```
$ cat /etc/delphix/platform
azure
```

Reply:
Yea, I think that'd be much bette.

I'm not sure how we'd implement that, though, give the current build system. How would we provide that file? I don't think we have a way to customize files provided by the debian package(s) based on the platform, do we?

Reply:
I think most of the `/var/opt/delphix/server.conf` is no longer needed but we could also append the platform information to that file.

Reply:
Just FYI, we're adding another occurance of `dpkg-query` + `sed` to get the platform in [this review](http://reviews.delphix.com/r/46631); so this would bring the duplication to delphix-platform, upgrade scripts, and now the app-gate.

Reply:
In that review, it's also using `dpkg-query -Wf '${Version}' "delphix-entire-$PLATFORM"` to get the "appliance version". If we end up writing a script to query the "appliance platform", it might be worth also adding a script to query the "appliance version" and perhaps the "appliance variant". At that point though, it might be better left for a follow up change since there's some feature creep now.

Reply:
I've added a file `/etc/delphix/info/platform`. Let me know what you think

Reply:
FYI, my gcp changes will build off of this too.